### PR TITLE
Route installation

### PR DESCRIPTION
**Pull request by cprabha**
Hi,

I have added code for configuring network instance type , router id and also variable for network instance name.
Please could you review the changes.

Thanks,
Prabha

### CODE REVIEW

**Comment by cprabha** [comment]
Hi Likai,

Please could you let me know if you got a chance to review this pull request.

Thanks,
Prabha

File: feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go [hunk 1]
```diff
@@ -82,6 +82,7 @@ const (
 	asPathRepeatValue      = 3
 	aclStatement1          = "10"
 	aclStatement2          = "20"
+	NetworkInstance        = "DEFAULT"
```
Comment (liulk) [inline feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go:85]: The network instance name seems to be vendor specific. Arista uses lower case "default". We can parameterize the vendor specific behavior by adding a deviation flag:

https://github.com/openconfig/featureprofiles/blob/main/internal/deviations/deviations.go

```
var (
  DefaultNetworkInstance = flag.String("deviation_default_network_instance", "DEFAULT", "The name used for the default network instance for VRF.")
)
```

Would you mind also do a find and replace so that all occurrences of the default network instance name uses the value from the deviation flag consistently across the device tests?

(These are tests under ate_tests, otg_tests, or tests, but ignore the config gen tests.)

File: feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go [hunk 2]
```diff
@@ -127,6 +128,18 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 	dc.Interface(i2.GetName()).Replace(t, i2)
 }
 
+//Configure network instance
```
Comment (liulk) [inline feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go:131]: Please ensure there is a whitespace after `//`. Also, comments should follow godoc convention.

https://go.dev/blog/godoc

```
// configureNetworkInstance configures the network instance.
```

Actually, in this case the comment doesn't really add new information, so let's just remove the comment.

File: feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go [hunk 3]
```diff
@@ -127,6 +128,18 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
...
+func configureNetworkInstance(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	d := &telemetry.Device{}
+	ni1 := d.GetOrCreateNetworkInstance(NetworkInstance)
+	ni1.Type = telemetry.E_NetworkInstanceTypes_NETWORK_INSTANCE_TYPE(*ygot.Int64(1))
```
Comment (liulk) [inline feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go:136]: Use the enum for the default network instance type, as opposed to a magic value?

File: feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go [hunk 4]
```diff
@@ -127,6 +128,18 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 	dc.Interface(i2.GetName()).Replace(t, i2)
 }
 
+//Configure network instance
+func configureNetworkInstance(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	d := &telemetry.Device{}
```
Comment (liulk) [inline feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go:134]: I know it's a little verbose, but how about constructing the network instance struct directly?

ni := &telemetry.NetworkInstance{
  Name: ...,
  Type: ...,
  RouterId: ...,
}

File: feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go [hunk 5]
```diff
@@ -127,6 +128,18 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
...
+	ni1 := d.GetOrCreateNetworkInstance(NetworkInstance)
+	ni1.Type = telemetry.E_NetworkInstanceTypes_NETWORK_INSTANCE_TYPE(*ygot.Int64(1))
+
+	ni1.RouterId = ygot.String(dutDst.IPv4)
```
Comment (liulk) [inline feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go:138]: Why does the default NI need a router ID?

File: topologies/atedut_3.testbed [hunk 6]
```diff
@@ -0,0 +1,47 @@
+# proto-file: github.com/openconfig/ondatra/blob/main/proto/testbed.proto
```
Comment (liulk) [inline topologies/atedut_3.testbed:1]: Revert this file? Those who need the 3 port variant should just use atedut_4. Also please revert the changes to file mode in the other *.binding and *.testbed files.

Please rebase to the latest main. I added some file mode checks.

**Review by liulk — CHANGES_REQUESTED** [review]

File: feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go (hunk 1 above)
Comment (cprabha) [reply feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go:85]: Hi Likai,

Thanks for the reviewing the changes.
I have updated deviations.go file . So in my script should i use variable "DefaultNetworkInstance " ? and initialize the value to "DEFAULT" as well? please could you clarify.

Comment (liulk) [reply feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go:85]: Yes, in your code, use *deviations.DefaultNetworkInstance, and initialize the value to DEFAULT (see my code snippet above).

**Comment by liulk** [comment]
Please remember to revert the file mode changes.

File: feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go (hunk 1 above)
Comment (cprabha) [reply feature/bgp/policybase/ate_tests/route_installation_test/route_installation_test.go:85]: Got it, Thanks Likai. 
I have updated the script with *deviations.DefaultNetworkInstance and validated it.
After this PR, will work on other test.go files.

**Comment by liulk** [comment]
I don't see any updates to the PR. Did you push the latest commit?

**Comment by cprabha** [comment]
Hi Likai,
I have not pushed changes yet, updating configureNetworkInstance to struct. 
I will push all the changes today shortly.

Thanks,
Prabha

**Review by liulk — APPROVED** [review]

File: internal/deviations/deviations.go [hunk 7]
```diff
@@ -28,4 +28,7 @@ var (
 
 	AggregateAtomicUpdate = flag.Bool("deviation_aggregate_atomic_update", true,
 		"Device requires that aggregate Port-Channel and its members be defined in a single gNMI Update transaction at /interfaces; otherwise lag-type will be dropped, and no member can be added to the aggregate (b/201574574)")
+
+        DefaultNetworkInstance = flag.String("deviation_default_network_instance", "DEFAULT", "The name used for the default network instance for VRF.")
```
Comment (liulk) [inline internal/deviations/deviations.go:32]: `go fmt` indicates that the indent should be a tab, not whitespace.

**Review by liulk — CHANGES_REQUESTED** [review]

**Comment by cprabha** [comment]
Looks like i need to fix deviations.go file. will fix it.

**Review by liulk — APPROVED** [review]

**Comment by cprabha** [comment]
Thanks Likai.
